### PR TITLE
add missing context manager release, fix memory leak

### PR DESCRIPTION
**Pull request by vikstrous**
Signed-off-by: Viktor Stanchev <me@viktorstanchev.com>

### CODE REVIEW

File: registry/handlers/app.go [hunk 1]
```diff
@@ -640,6 +640,7 @@ func (app *App) dispatcher(dispatch dispatchFunc) http.Handler {
 		}
 
 		context := app.context(w, r)
+		defer defaultContextManager.release(context)
```
Comment (dmcgowan) [inline registry/handlers/app.go]: Is the `*http.Request` always different than the request object from `(*App).ServerHTTP`?

Comment (vikstrous) [inline registry/handlers/app.go]: Hmm... good question. We don't track number of references, so this might have unexpected consequences...

Comment (dmcgowan) [inline registry/handlers/app.go]: It should be safe to do multiple releases, my main concern is that if there was a memory leak discovered are we sure this fixes it.

Comment (stevvooe) [inline registry/handlers/app.go]: The problem with this approach is that in 1.7+, the address of the request object can change between handlers, making the technique used for the context manager invalid. The other PR is much safer.

Comment (vikstrous) [inline registry/handlers/app.go]: Ouch. I didn't know that. I guess we don't have much of a choice then.

Comment (stevvooe) [inline registry/handlers/app.go]: @vikstrous Yea, unfortunately, based on what I discovered in the other patch, this addition to `net/http` _kind of_ backwards incompatible. Although, I'm not complaining, as the result is much better than it was.

**Pull request closed without merging**